src/terminator/mod.rs

```
@@ -567,17 +676,103 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
+                } else {
+                    info!("Ignored output to FD {}", fd);
+                    n as isize // pretend it all went well
+                }; // now result is the value we return back to the program
+                self.write_primval(dest, PrimVal::Bytes(result as u128), dest_ty)?;
             }
 
-            // unix panic code inside libstd will read the return value of this function
-            "pthread_rwlock_rdlock" => {
+            // Some things needed for sys::thread initialization to go through
+            "signal" | "sigaction" | "sigaltstack" => {
                 self.write_primval(dest, PrimVal::Bytes(0), dest_ty)?;
             }
 
+            "sysconf" => {
+                let name = self.value_to_primval(args[0], usize)?.to_u64()?;
+                trace!("sysconf() called with name {}", name);
+                let result = match name {
+                    30 => 4096, // _SC_PAGESIZE
+                    _ => return Err(EvalError::Unimplemented(format!("Unimplemented sysconf name: {}", name)))
+                };
+                self.write_primval(dest, PrimVal::Bytes(result), dest_ty)?;
+            }
+
+            "mmap" => {
+                // This is a horrible hack, but well... the guard page mechanism calls mmap and expects a particular return value, so we give it that value
+                let addr = args[0].read_ptr(&self.memory)?;
+                self.write_primval(dest, PrimVal::Ptr(addr), dest_ty)?;
+            }
+
+            // Hook pthread calls that go to the thread-local storage memory subsystem
+            "pthread_key_create" => {
+                let key_ptr = args[0].read_ptr(&self.memory)?;
+                
+                // Extract the function type out of the signature (that seems easier than constructing it ourselves...)
+                let dtor_ptr = args[1].read_ptr(&self.memory)?;
+                // TODO: The null-pointer case here is entirely untested
```

I'm not sure how to test it. Just panic so we see when we trigger it, instead of potentially doing something wrong silently